File: builtin/fs.lua
function (_ENV)
local string, math, print, type, pairs = nil, nil, nil, nil, nil;
local _module = _obj({exports=_obj({})}); local exports, module = _module.exports, _module;

(exports).readFileSync = (function (this)
if true then return (""); end;
end);
(exports).readdirSync = (function (this, path)
local ptr, dir, dirs = ptr, dir, dirs;
ptr = (function () local base, prop = (ffi).C, "tm_fs_dir_open"; return base[prop](base, path); end)();
dir = nil;
if _truthy((ptr == undefined)) then
_error((("ENOENT: Could not open ") + path))
end
dirs = _arr({});
while ((function () local _r = (function () local base, prop = (ffi).C, "tm_fs_dir_next"; return base[prop](base, ptr); end)(); dir = _r; return _r; end)() ~= undefined) do

if dirs:push(ffi:string(dir)) then end;

end
if (function () local base, prop = (ffi).C, "tm_fs_dir_close"; return base[prop](base, ptr); end)() then end;
if true then return dirs; end;
end);

return _module.exports;
end

File: builtin/fs.js
var tm = process.binding('tm');

exports.readFileSync = function (pathname) {
  var fd = tm.fs_open(pathname, tm.OPEN_EXISTING | tm.RDONLY);
  var res = [];
  while (true) {
    if (tm.fs_readable(fd) == 0) {
      var len = 100;
      var buf = tm.fs_read(fd, len);
      if (buf.length > 0) {
        res.push(buf);
      }
      if (buf.length < len) {
        break;
      }
    }
  }
  return res.join('');
};

exports.readdirSync = function (pathname) {
  var dir = tm.fs_dir_open(pathname);
  if (dir == undefined) {
    throw 'ENOENT: Could not open ' + pathname;
  }
  var entries = [];
  while ((ent = tm.fs_dir_read(dir)) != undefined) {
    if (ent != '.' && ent != '..') {
      entries.push(ent);
    }
  }
  tm.fs_dir_close(dir);
  return entries;
};

exports.readdir = function (pathname, next) {
  setImmediate(function () {
    next(null, exports.readdirSync(pathname));
  })
}

exports.readFile = function (pathname, next) {
  setImmediate(function () {
    next(null, exports.readFileSync(pathname));
  });
};